### PR TITLE
Creating dockers to run asgard

### PR DESCRIPTION
**Pull request by AurelienLP**
Inspired from @kinnou02 knowledge and https://medium.com/@mattias.holmlund/building-the-perfect-data-container-2df6bb3528df

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -47,3 +47,17 @@ pre-commit run <hook_id>
 ```
 
 The CI will fail if the code is not correctly formated so be careful !
+
+### DOCKER
```
Comment (xlqian) [inline README.md]: :+1:

File: README.md [hunk 2]
```diff
@@ -47,3 +47,17 @@ pre-commit run <hook_id>
...
+
+There are 3 differents docker images in the project :
+
+    - asgard-dev : compiles and installs a specific release of Valhalla and contains all the librairies to run Asgard.
```
Comment (xlqian) [inline README.md]: the release version is actually hard coded in Dockerfile, can you expose it to arg?

File: README.md [hunk 3]
```diff
@@ -47,3 +47,17 @@ pre-commit run <hook_id>
...
+    - asgard-data : creates the configuration file to run asgard and the tiles of a specific pbf. By default it contains the tiles of the whole France.
+    - asgard-prod : compiles and run Asgard with the data of a asgard-data container.
```
Comment (xlqian) [inline README.md]: we may want also to be able to choose the version of asgard

File: README.md [hunk 4]
```diff
@@ -47,3 +47,17 @@ pre-commit run <hook_id>
...
+
+To run Asgard with docker :
+```bash
+docker create --name asgard-data navitia/asgard-data:latest
+docker run --volumes-from asgard-data --rm -it -p 6000:6000/tcp navitia/asgard-prod:latest
+```
```
Comment (xlqian) [inline README.md]: As we discusses, can you add a little graph to talk about little more on the "advanced" usage? I mean use "asgard-data" with tags which are names of dataset

File: docker/asgard-data/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+
+# Download the pbf given in argument
+ARG pbf_url=http://download.geofabrik.de/europe/france-latest.osm.pbf
+RUN wget --progress=bar:force:noscroll $pbf_url \
+  && mkdir -p valhalla_tiles \
+  && valhalla_build_config --mjolnir-tile-dir ${PWD}/valhalla_tiles --mjolnir-tile-extract ${PWD}/valhalla_tiles.tar --mjolnir-timezone ${PWD}/valhalla_tiles/timezones.sqlite --mjolnir-admin ${PWD}/valhalla_tiles/admins.sqlite > valhalla.json \
+  && valhalla_build_tiles -c valhalla.json *.pbf \
+  && find valhalla_tiles | sort -n | tar cf valhalla_tiles.tar --no-recursion -T -
+
+FROM scratch
```
Comment (xlqian) [inline docker/asgard-data/Dockerfile]: :heart:

File: README.md (hunk 2 above)
Comment (AurelienLP) [reply README.md]: We could easily do that indeed ! Let's see what the others think about that but I think it is a good idea :)

File: README.md (hunk 3 above)
Comment (AurelienLP) [reply README.md]: Yes indeed !

File: README.md (hunk 4 above)
Comment (AurelienLP) [reply README.md]: Will do that.

Comment (AurelienLP) [reply README.md]: @xlqian Just added it ! What do you think ?

File: README.md (hunk 2 above)
Comment (kinnou02) [reply README.md]: it's not that useful to change the valhalla version at build time.
We want to keep a hand on what version is used.
After all valhalla is a dependency and I prefer to pin version of dependencies

File: README.md (hunk 3 above)
Comment (kinnou02) [reply README.md]: for that the best way is to create a tag in dockerhub for each version of asgard

File: README.md (hunk 4 above)
Comment (xlqian) [reply README.md]: :+1:

File: docker/asgard-app/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,15 @@
+FROM navitia/asgard-build-deps:latest
+
+RUN apt-get update && apt-get install -y --no-install-recommends
```
Comment (TeXitoi) [inline docker/asgard-app/Dockerfile]: Useful? if not, l12 must also be useless.

Comment (AurelienLP) [reply docker/asgard-app/Dockerfile]: Done

File: docker/asgard-build-deps/Dockerfile [hunk 7]
```diff
@@ -41,21 +41,22 @@ RUN apt-get update && apt-get install -y --no-install-recommends git \
       cmake \
       libzmq3-dev \
       ca-certificates \
-      wget \
-      clang-format-4.0 \
-  && git clone --depth=1 --recursive https://github.com/valhalla/valhalla.git libvalhalla \
-  && mkdir -p libvalhalla/build \
+      clang-format-4.0
+
+RUN git clone --depth=1 --recursive --branch 3.0.2 https://github.com/valhalla/valhalla.git libvalhalla
```
Comment (kinnou02) [inline docker/asgard-build-deps/Dockerfile]: Multiple `RUN` will create multiple layer that will increase the resulting image size

**Pull request merged**